phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java

```diff
@@ -122,7 +123,38 @@
             ColumnFamilyDescriptorBuilder.KEEP_DELETED_CELLS,
             ColumnFamilyDescriptorBuilder.REPLICATION_SCOPE);
 
-    public static boolean areClientAndServerCompatible(long serverHBaseAndPhoenixVersion) {
+    public static class ClientServerCompatibility {
+
+        long serverVersion;
+        StringBuilder errorCode;
+        boolean isCompatible;
+
+        ClientServerCompatibility(long serverVersion) {
+            this.serverVersion = serverVersion;
+            this.errorCode = new StringBuilder();
```

Comment:
Why is `errorCode` a StringBuilder? SQLExceptionCode error code types are integers.